src/NHibernate/Proxy/StaticProxyFactory.cs

```diff
@@ -43,9 +44,16 @@ public override INHibernateProxy GetProxy(object id, ISessionImplementor session
 
 		public override object GetFieldInterceptionProxy(object instanceToWrap)
 		{
-			var factory = new ProxyFactory();
-			var interceptor = new DefaultDynamicLazyFieldInterceptor();
-			return factory.CreateProxy(PersistentClass, interceptor, typeof(IFieldInterceptorAccessor));
+			var cacheEntry = new ProxyCacheEntry(IsClassProxy ? PersistentClass : typeof(object), System.Type.EmptyTypes);
```

Comment:
`IsClassProxy ? PersistentClass : typeof(object)` has to be reverted to `PersistentClass`: the field interception proxy does not handle being an interface proxy. (I will add a test, this is lacking.)

(Optionally, it may be attempted here to handle being an interface proxy. But the `INHibernateProxy` interface will have to be filtered out of `Interfaces`. And the support would only be partial, as there will be no way of handling interface methods, but only properties. So I do not think it should be done.)